geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseConnectionPool.scala

```diff
@@ -27,24 +27,33 @@ object HBaseConnectionPool extends LazyLogging {
 
   private var userCheck: Option[User] = _
 
-  // add common resources from system property
-  private val configuration = withPaths(HBaseConfiguration.create(), HBaseDataStoreFactory.ConfigPathProperty.option)
-
-  private val configCache = Caffeine.newBuilder().build(
+  private val zkConfigCache = Caffeine.newBuilder().build(
     new CacheLoader[(Option[String], Option[String]), Configuration] {
       override def load(key: (Option[String], Option[String])): Configuration = {
         val (zookeepers, paths) = key
+        val configuration = withPaths(HBaseConfiguration.create(), HBaseDataStoreFactory.ConfigPathProperty.option)
```

Comment:
can we move this back outside the cache, or will that mess something up? that way we don't have to re-load the base config for different params.

Reply:
We can move it out only for the zkConfigCache, the confConfigCache has to hand it the incoming configuration. I moved it in here since I thought it made it less confusing but we can move it. out.